src/components/login/SignUp.tsx

```diff
@@ -9,57 +14,140 @@ import {
   CustomText,
   PreviousPageBtn,
 } from "../CustomMUI/CustomUI";
-
+import { Visibility, VisibilityOff } from "@mui/icons-material";
+import { visiblePasswordState } from "../../types";
+import { validatePassword } from "../../server/utils/password";
+import DoNotDisturbAltIcon from "@mui/icons-material/DoNotDisturbAlt";
 type signUpProp = {
-  userClickedBack: (value: boolean) => void;
+  goBack: (value: boolean) => void;
 };
-const SignUp: React.FC<signUpProp> = ({ userClickedBack }) => {
+
+const SignUp: React.FC<signUpProp> = ({ goBack }) => {
   const [userName, setUserName] = useState<string>("");
+  const [strongPassword, setStrongPassword] = useState<boolean>(true);
+  const [usernameExists, setUsernameExists] = useState<boolean>(false);
   const [userPassword, setUserPassword] = useState<string>("");
   const [firstName, setFirstName] = useState<string>("");
   const [lastName, setLastName] = useState<string>("");
+  const [values, setValues] = useState<visiblePasswordState>({
+    showPassword: false,
+  });
 
-  const handleSignUp = async () => {
-    await fetch(`/api/account`, {
-      method: "POST",
+  const handleSignUp = async (e: FormEvent) => {
+    e.preventDefault();
+    if (strongPassword) {
+      const res = await fetch(`/api/useraccount`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          username: userName,
+          userpassword: userPassword,
+          firstName: firstName,
+          lastName: lastName,
+        }),
+      });
+      if (res.status === 200) {
+        const timer = setTimeout(() => {
+          goBack(true);
+        }, 300);
+        return () => clearTimeout(timer);
+      } else {
+        setUsernameExists(true);
+      }
```

Comment:
This is a ver optimistic thing to do. What if the status returned was 500 (internal server error)? It's not 200, so it will result in an error message being shown telling the user that the username was taken, even though there was actually something else that went wrong.

For error messages, it's a good idea to return precise error codes and check for them, e.g. 409 (conflict) when a username is already taken, and to only show that error message for status 409. For other non-200 codes, a more general error message could be shown.